src/cartesian/Area.tsx

```diff
@@ -257,6 +261,10 @@ function StaticArea({
 }) {
   const { layout, type, stroke, connectNulls, isRange, ...others } = props;
 
+  const { id, ...propsWithoutId } = props;
```

Comment:
Doesn't this mean that id won't make it to the dom? Not great when its user specified

Reply:
The id was applied to repeat elements (like dots) so I have to filter it away. It should still remain on the top level.

Reply:
Not sure if we have tests for that. I suppose I could add those first

Reply:
https://github.com/recharts/recharts/pull/6109